tests/test_context.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+from plumbum import local
+
+import copier
+
+from .helpers import build_file_tree, git_save
+
+
+@pytest.mark.parametrize("operation", ("recopy", "update"))
+def test_operation_in_context_matches(
+    operation: str, tmp_path_factory: pytest.TempPathFactory
+) -> None:
+    """
+    Ensure that the _copier_conf.operation context variable is set
+    as expected during template rendering.
+    """
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+    with local.cwd(src):
+        build_file_tree(
+            {
+                # Ensure the file is regenerated on update.
+                "copier.yml": "_skip_if_exists: [foo]",
+                "{{ _copier_conf.answers_file }}.jinja": "{{ _copier_answers|to_yaml }}",
+                "foo.jinja": "{{ _copier_conf.operation }}",
+            }
+        )
+        git_save(tag="1.0.0")
+
+    copier.run_copy(str(src), dst, defaults=True, overwrite=True)
+    ctx_file = dst / "foo"
+    assert ctx_file.read_text() == "copy"
+    # Ensure the file is regenerated on update.
+    # If we left it, an update would detect custom changes
+    # that would be reapplied, i.e. an update would leave us with `copy`.
+    ctx_file.unlink()
+    with local.cwd(dst):
+        git_save()
+    getattr(copier, f"run_{operation}")(str(dst), overwrite=True)
+    expected = "copy" if operation == "recopy" else operation
+    assert ctx_file.read_text() == expected
```

Comment:
thought: So, let's say that we copy one template and render the value. The file will contain `copy`.

Now we run it again. The file contains `update`. The change is OK because the replay renders `copy` and detects the change.

However, what would happen if we run it yet again? The replay will render `copy`, but in reality it is `update`. Now the next one is also `update`.

Triple-thinking this... are we introducing here a way to make replays non-reproducible and thus maybe making the update algorithm unreliable? Because there's no way Copier could know if last play was a copy or an update.

CC @copier-org/maintainers. Should we just reject this feature to avoid shooting our own feet?

Reply:
Disregarding that `_copier_conf.operation` is not intended to be used for template content changes, this test for checking the value of `operation` only works because the file containing `{{ _copier_conf.operation }}`

1. is deleted
2. is marked as `skip_if_exists`, which means it will be regenerated during `update` if deleted.

What would happen without it being deleted is the following:

1. `copier copy` renders the template, it contains `copy`
2. `copier recopy` overwrites the template, it contains `copy`
3. `copier update` updates the template, it still contains `copy`. This is valid for each subsequent update.

Why (3):
During `_apply_update`, `old_copy` and `new_copy` are both rendered with `_copier_conf.operation == "update"`. The file contains `copy` though, which is detected as a custom change and reapplied.

`_copier_conf.operation` should be documented as a meta variable, intended for templating `copier.yml` configuration. It cannot be used to influence template contents.